UDPServer/ServerContextRepo.cs

```diff
@@ -0,0 +1,54 @@
+﻿using System.Threading.Tasks;
+using UDPCommGUI;
+using System.Linq;
+using System;
+
+namespace Sice.PoC.UDPServer
+{
+    public class ServerContextRepo
+    {
+        public ServerContext context { get; set; }
+        public ServerContextRepo(ServerContext context)
+        {
+            this.context = context;
+        }
+
+        public async Task<int> AddReceivedMessage(ReceivedMessage message)
+        {
+            context.ReceivedMessages.Add(message);
+            return await context.SaveChangesAsync();
+        }
+
+        public int GetControllerIDFromLogin(ClientLoginEvent credential)
+        {
+            var username = credential.Username;
+            var password = credential.Password;
+            var query = from login in context.Login
+                        where login.Username == username
+                        select login;
+            foreach (var items in query)
+            {
+                if (items.Username == username && BCrypt.Net.BCrypt.Verify(password, items.PasswordHash))
+                {
+                    return items.ControllerID;
+                }
+            }
```

Comment:
Suggest turning this into a LINQ query.

Ideally any business logic (example verification) belongs in a service. So the repository is only responsible for query logic and combining data. 

Nice use on the BCrypt.